File: Easy Problems/Two Sums/Python/Optimized.py
class Solution:
        def twoSum(self, nums: List[int], target: int) -> List[int]:
            num_map = {}
        
            for i, num in enumerate(nums):
                item = target - num
                if item in num_map:
                    return [num_map[item], i]
                num_map[num] = i

            return []

File: Easy Problems/Two Sums/Java/Optimized.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numMap = new HashMap<>();

        for(int i = 0; i < nums.length; i++)
        {
            int item = target - nums[i];
            if(numMap.containsKey(item))
            {
                return new int[] {numMap.get(item), i};
            }

            numMap.put(nums[i], i);
        }

        return new int[0];
    }
}
